src/main/scala/viper/server/VerificationWorker.scala

```diff
@@ -72,13 +60,40 @@ class VerificationWorker(private val reporterActor: ActorRef,
   }
 
   private var backend: ViperBackend = _
+  implicit val executionContext = ExecutionContext.global
+
+  // Implementation of the Reporter interface used by the backend.
+  class ActorReporter(private val actor_ref: ActorRef,
+                      val tag: String) extends viper.silver.reporter.Reporter {
+
+    val name = s"ViperServer_$tag"
+    var current_offer: Future[QueueOfferResult] = null
+
+    /** Sends massage to the attached actor.
+      *
+      * The actor receving this message offer it to a queue. This offering returns a Future, which will eventually
+      * indicate whether or not the offer was successful. This method is blocking, as it waits for the successful
+      * completion of such an offer.
+      * */
+    def report(msg: reporter.Message): Unit = {
+      implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
+      val answer = actor_ref ? ReporterProtocol.ServerReport(msg)
+      current_offer = answer.flatMap({
+        case res: Future[QueueOfferResult] => res
+      })
+      while(current_offer == null || !current_offer.isCompleted){
+        Thread.sleep(10)
```

Comment:
Same as above.

Reply:
Is this then meant to be an adjustable variable for clients? Because parametrising this feels like exposing fairly low level aspects of the server's behaviour for not much reason. In particular "SourceQueue_offer_backOff_interval" doesn't seem to be a very intuitive parameter for a client to set. 😄 

But implementation-wise it should be very much feasible.

Reply:
Definitely not for the end users, but for whoever might want to experiment with the infrastructure — could be useful. IF you think it's not worth making this interval configurable, please add a comment into the code to explain what it means.